### DISCUSSION TITLE
Socket.io-client + SignalR(.net) server

### DISCUSSION THREAD

**Discussion by themod161** [socketio/socket.io#4956] [Q&A] [answered]
Hi, how can I friend SignalR (C#) on server and socket.io-client in reactjs? If you can with the code 🙏

**Comment by darrachequesne** [comment]
Hi! Unfortunately, I don't think this is possible, as the SignalR protocol is quite different from the Socket.IO protocol.

**Reply by themod161** [reply]
What about using websockets? It's possible?

**Reply by darrachequesne** [reply]
Sorry, I'm not sure to understand. Could you please explain your use case?

**Reply by themod161** [reply]
Mb I don't know how socket-io working.
I found this page - https://socket.io/docs/v3/client-initialization/#transports 

Can I connect to signalr by websocket. If i can, then how? Else, what it's analog to socket.io (server) in c#?

**Reply by darrachequesne** [reply, marked as answer]
You can find more information about how Socket.IO works here: https://socket.io/docs/v4/how-it-works/

> Can I connect to signalr by websocket?

Yes, I guess you should be able to connect with a plain WebSocket connection, but you will certainly have to format the messages in a specific way. I am no SignalR expert though.

> Else, what it's analog to socket.io (server) in c#?

No, unfortunately there is currently no server implementation in c# (only client).
